### PR TITLE
Comments for next semester's dev team, implement clearer naming conventions

### PR DESCRIPTION
**Pull request by lixitrixi**
Michael and I discussed possible Network structures, and added comments to reflect ARTwarp goals for those working on the project next year.

### CODE REVIEW

**Comment by olexandr-konovalov** [comment]
@lixitrixi please investigate failing unit tests

**Comment by lixitrixi** [comment]
I will wait for @michaelstanway's recent pull request to be merged, then add the comments from this commit to the changed files to avoid unnecessary conflicts.

@olexandr-konovalov

**Comment by harvard-tham** [comment]
the error is a simple change in the unit test from weight to weights. Can i checkout this or does it have to be Felix that changes it.

File: Category.m [hunk 1]
```diff
@@ -4,27 +4,34 @@
...
+        size % The number of contours in this category
     end
     
     methods
-        function cat = Category(contours)
+        function cat = Category(size, reference)
             % Construct a new category from the given array, which must
             % contain at least one contour
-            cat.contours = contours;
-            cat.average  = ContourFactory.getAverage(contours);
```
Comment (harvard-tham) [inline Category.m:17]: `ContourFactory` doesn't have `.getAverage( )` yet. Are you sure we want to have that in there?

File: Network.m [hunk 2]
```diff
@@ -4,14 +4,15 @@
     %   repertoire through refinement with training data
     
     properties
-        weight
+        weights
```
Comment (harvard-tham) [inline Network.m:7]: this is what caused the unit test to fail

**Review by harvard-tham — CHANGES_REQUESTED** [review]
Update the Network unit test to match the updated attribute

**Comment by harvard-tham** [comment]
found out how to push to your fork, should work now

**Review by harvard-tham — APPROVED** [review]

**Review by olexandr-konovalov — CHANGES_REQUESTED** [review]
Thanks, @harvard-tham, for fixing tests.

@lixitrixi you should minimise putting unrelated things into same PR. CLI stuff could be easily done as a separate PR and one would not hold the other. It would also ease rebasing. For example, I could squash your changes and @harvard-tham's fix, to make a commit atomic. But now there is your CLI commit in the middle, which also needs fixing the commit message. It has unmatched `)` but is also written not clearly - what do you mean by " must be rewritten to work with new backend"? Is that work in progress? Moreover should be separate PR. Or is it for the current backend and later would have to be rewritten - in this case that should rather be said in the further text of the commit message, rather then in the 1st line.

**Comment by olexandr-konovalov** [comment]
@lixitrixi requested changes - would be nice to address them and merge this ASAP. I don't think #50 is quite ready yet.

By the way, use the `#<PR number>` syntax for cross-referencing, rather than saying "@michaelstanway's recent pull request".

**Comment by olexandr-konovalov** [comment]
@lixitrixi P.S. and of course if the CLI code is runnable, the CI already is ready to run it on GitHub actions. If not runnable, then not mergeable. So that should be really a separate PR. Would you like me to refactor it, or would you do it yourself? You need to create a new branch, cherry-pick the CLI commit there, then go back to the branch for this PR and rebase it to remove the CLI commit. BTW, you again submit a PR from your `rewrite`, forgetting to create a feature branch - one more argument to push for merging rewrite into main, but we can't do it as we are keeping adding stuff and it becomes hard to sync moving targets.

**Comment by olexandr-konovalov** [comment]
What's the plan here? Unless anyone wishes to do this, I will take out @lixitrixi's CLI into a separate unmerged PR, and squash and merge the rest.

**Comment by lixitrixi** [comment]
@olexandr-konovalov I apologize for the problems, I will CLI code into a separate PR as soon as I can. I am currently settling back into life at home, but I will find some time this week! I the CLI is completely separate, so it will be easy to make them separate.

**Comment by olexandr-konovalov** [comment]
@lixitrixi thanks, sounds good, take your time then - will look here in a couple of weeks!

**Comment by lixitrixi** [comment]
@olexandr-konovalov Sorry for the extended hiatus...

I have removed CLI-related files from this PR, and will add them to a separate PR after refactoring and making the code more complete. Just to confirm, you mean that I should create a separate feature branch on my fork for CLI code? As in: dolphin-acoustics-vip:rewrite <- lixitrixi:CLI-code (or similar) ?

**Comment by lixitrixi** [comment]
The PR adding CLI code is ready on my fork, just waiting on this to be merged since that PR currently includes all the changes in this one as well.

**Comment by olexandr-konovalov** [comment]
Thanks @lixitrixi - there could be a better way to refactor the PR then deleting files, but to save time I have just squashed and merge it instead of another refactoring.

**Comment by olexandr-konovalov** [comment]
@lixitrixi now #46 merged and you can proceed with adding CLI. However, I am afraid that it's now based on the wrong branch. From now on, contributions are done not to the `rewrite` branch but to `main` (easier). For any change, pull the latest state of `main`, create a branch off it, add a change, submit a PR from it.

**Pull request merged**